example/src/main/java/org/wordpress/android/fluxc/example/ui/products/WooProductsFragment.kt

```diff
@@ -402,9 +402,14 @@ class WooProductsFragment : Fragment() {
         fetch_global_addons_groups.setOnClickListener {
             selectedSite?.let { site ->
                 coroutineScope.launch {
-                    addonsStore.fetchAllGlobalAddonsGroups(site).run {
-                        error?.let { prependToLog("${it.type}: ${it.message}") }
-                            prependToLog("Global addons: ${this.model}")
+                    addonsStore.fetchAllGlobalAddonsGroups(site).let { result ->
+                        prependToLog(
+                                if (result.error != null) {
+                                    "${result.error.type}: ${result.error.message}"
+                                } else {
+                                    "Global addons fetch successful"
+                                }
+                        )
```

Comment:
Do you think it would be more direct if we operate the error directly? Something like the previous code was, maybe like this:

```suggestion
                    addonsStore.fetchAllGlobalAddonsGroups(site).error
                            ?.let { prependToLog("${it.type}: ${it.message}") }
                            ?: prependToLog("Global addons fetch successful")
```

No strong opinions about this, just a nitpick.

Reply:
Looks good, neater - thanks!